### DISCUSSION TITLE
Hooks error when using whyDidYouRender

### DISCUSSION THREAD

**Discussion by callumbooth** [vercel/next.js#18453] [Help]
# Bug report

## Describe the bug

When following the with-why-did-you-render example, having a page using a hook and making a change to the _app file will cause the following `Error: Rendered more hooks than during the previous render`. I wasn't sure if this issue should exist here or on the whyDidYouRender repo

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to https://codesandbox.io/s/flamboyant-pascal-qhndz
2. Open up the _app file
3. Edit the text in above the layout component

## Expected behavior

The number of hooks called should be the same and no error should be displayed

## System information

- OS: macOS
- Browser (if applies) chrome
- Version of Next.js: 9.5.4 and 10.0.0
- Version of Node.js: 10.21.0

**Comment by jamesmosier** [comment]
Hi @callumbooth. I believe this is just the nature of how `why-did-you-render` is used. Since `why-did-you-render` is used outside of the function component, fast refresh is viewing the [hooks in `why-did-you-render`](https://github.com/welldone-software/why-did-you-render/blob/78c9eb9f263f2a90ccc96908bcce5b86f58824eb/src/whyDidYouRender.js#L19) as outside of the tree and therefore cannot keep track of their render order.

Fast refresh only preserves state in [function components](https://nextjs.org/docs/basic-features/fast-refresh#tips) so this likely falls outside of that requirement.
